docs/build-provenance-for-all-registries.md

```diff
@@ -0,0 +1,104 @@
+# Build Provenance for All Registries
+
+There’s a new security capability that registries can offer - build provenance - which verifiably links a package back to its source code and build instructions.
+
+Build provenance starts with the build system making information about source code and build instructions available in a non-falsifiable way. During a build, that information is sent to a public-good signing service called [Sigstore](https://www.sigstore.dev/how-it-works), which provides a public transparency log for auditing purposes as well as enabling signing without requiring individual users to manage keys. The signed build provenance is then sent to a registry on package upload, and served to users when they consume a package. In this document, we describe the design decisions and high-level implementation for adding build provenance to npm, to show other registries how they could also support this capability.
+
+Before we talk about the implementation, let's talk through the design decisions. **The goal is to have packages be transparent about their contents.** Note that we aren’t saying a package with provenance is _inherently_ secure or trustworthy, just that non-falsifiable links to source code and build instructions exist for consumers to follow and make their own determination, either through manual inspection or by evaluating provenance with their own policies. This is a stronger guarantee than information optionally and manually being supplied by maintainers.
+
+For provenance to provide strong security guarantees, it requires using a build system that can:
```

Comment:
This feels like the start of a `## Package Building` section

Reply:
I reworked the introduction based on your feedback to have more explicit sections.